### PR TITLE
Integrate ONNX-Chainer to Chainer repository

### PR DESCRIPTION
**Pull request by disktnk**
Merge ONNX-Chainer to Chainer mainstream

- don't have to install add-on package, like `$ pip install onnx-chainer`
    - but user have to install `onnx` and `packaging` manually
- to catch up latest Chainer modification

Merging logic is same as ChainerMN (#5226), user can `import onnx_chainer` after installing Chainer.

1. Copied [ONNX-Chainer](https://github.com/chainer/onnx-chainer), based https://github.com/chainer/onnx-chainer/commit/f3443bb4c9ae3657c33590e72b4b0b8e19d3a74e
    1. onnx_chainer/onnx_chainer --> onnx_chainer
    1. tests --> tests/onnx_chainer_tests
    1. docs --> docs/source/onnx_chainer, deployed to personal account https://disktnk-chainer.readthedocs.io/en/latest/onnx_chainer/index.html
1. Add CI project, current ONNX-Chainer's TravisCI is merged to flexci
    * onnxchainer.py37.gpu
    * onnxchainer.py37.cpu
    * onnxchainer.py36.cpu
    * onnxchainer.py36.cpu.onnx14
    * onnxchainer.py35.cpu
    * onnxchainer.win.py37.gpu

Some files are modified to install ONNX-Chainer module, gathered on this commit https://github.com/chainer/chainer/commit/3e875d823ede06f6ed5f1f4134701eeda16c3077

### CODE REVIEW

**Comment by toslunar** [comment]
Jenkins & flexCI, test this please.

File: onnx_chainer/export.py [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+
+    ``y = model(*args)``
+
+    ``external_converters`` and ``external_opset_imports`` are for external
+    custom operator. When some ~chainer.FunctionNode are expected to convert to
+    own customized operator, set converter function with ~chainer.FunctionNode
+    name.
+
+    >>> import onnx
+    >>> def custom_converter(param):
```
Comment (toslunar) [inline onnx_chainer/export.py]: doctest failed here.  It seems `...` (in place of `>>>`) should be used on continuation lines.  E.g.
```
    >>> def custom_converter(param):
    ...     return onnx.helper.make_node(
    ...         'CustomizedRelu', param.input_names, param.output_names,
    ...         domain='chainer'),
```

Comment (disktnk) [reply onnx_chainer/export.py]: Thanks, got it! I fixed them.

**Comment by toslunar** [comment]
Jenkins & flexCI, test this please.

**Comment by toslunar** [comment]
Jenkins & flexCI, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**